### PR TITLE
Refactor board directory.

### PR DESCRIPTION
**Pull request by manoharahk**
Move board directories from arch to os/board

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b8ea0208d4ee32258d74c990eb040067cba48872] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b8ea0208d4ee32258d74c990eb040067cba48872] - Code Rule Check Result:
os/board/lm3s6965-ek/src/up_boot.c:118: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/board/lm3s6965-ek/src/up_boot.c:120: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/lm3s6965-ek/src/up_boot.c:121: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/lm3s6965-ek/src/up_boot.c:122: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/lm3s6965-ek/src/up_boot.c:123: ERROR: [IDT_M_TAB] please, no space before tabs
os/board/lm3s6965-ek/src/up_boot.c:124: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [f7e747d11e37e39282baefc7be0b0f9cf5e425c9] - Code Rule Check OK.

**Comment by manoharahk** [comment]
Doc update in https://github.com/Samsung/TizenRT/blob/master/docs/HowToAddnewBoard.md is pending..

**Comment by sunghan-chang** [comment]
@manoharahk Could you let me know Doc. schedule?

**Comment by manoharahk** [comment]
I will try to update doc by today.

**Comment by seinfra** [comment]
Target : [e365b208044617db4d8ab5003b6609be9f2f1806] -

**Comment by seinfra** [comment]
Target : [99f57d3faf22c54a7b0f0859b8ba074d8b16a842] - Code Rule Check OK.

File: os/Kconfig [hunk 1]
```diff
@@ -439,6 +439,10 @@ menu "Hardware Configuration"
 source arch/Kconfig
 endmenu
 
+menu "Board Selection"
```
Comment (sunghan-chang) [inline os/Kconfig]: arch/Kconfig's title is "Hardware Selection". Let's change it as "Chip Selection".

Comment (manoharahk) [reply os/Kconfig]: Ok

Comment (sunghan-chang) [reply os/Kconfig]: And then let's remove arch/Kconfig and rename Kconfig.chip to Kconfig.

Comment (manoharahk) [reply os/Kconfig]: Ok

**Comment by seinfra** [comment]
Target : [2dfba774c649f70bb2999296c531f0f0c0e8a5d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2dfba774c649f70bb2999296c531f0f0c0e8a5d2] - Code Rule Check OK.

**Pull request merged**